internal/controller/instaslice_controller.go

```diff
@@ -360,6 +348,11 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 								return ctrl.Result{}, err
 							}
 							resultRemove, err := r.removeInstaSliceFinalizer(ctx, req)
+							r.CleanupOrphanedAllocations(ctx, &instasliceList)
+							// update DeployedPodTotal Metrics by setting value to 0 as pod allocation is deleted and pod is no loger consuming slices
+							r.UpdateDeployedPodTotalMetrics(string(allocation.Nodename), allocation.GPUUUID, allocRequest.PodRef.Namespace, allocRequest.PodRef.Name, allocRequest.Profile, 0)
+							//update compatible profiles metrics
+							r.UpdateCompatibleProfilesMetrics(instaslice, instaslice.Name)
 							if err != nil {
 								return resultRemove, err
 							}
```

Comment:
I think we are at a point where we should add metrics update to a separate reconcile loop, the current code changes will cause the core reconcile loop to be retired for an error in metrics update, that does not seem to be correct. This will cause latency issues in the system for scheduling the next slice.

Reply:
The code needs to be reordered, err is not from metrics which confused me as git tries to collapse unchanged code.